### ISSUE TITLE
linux-poplar fails in thud branch

### ISSUE THREAD

**Issue by mirzak** [96boards/meta-96boards#310]
Hi,

Today I tested running a `poplar` board using the `thud` branch of the oe-rpb-manifest[](https://github.com/96boards/oe-rpb-manifest) but I am getting a lot of errors during the `linux-poplar` build due to usage of gcc8,

example of error:

```
drivers/msp/pvr/drv_pvr_intf.c:255:56: error: argument to 'sizeof' in 'strncpy' call is the same expression as the source; did you mean to use the size of the destination? [-Werror=sizeof-pointer-memaccess]
```

I tried fixing a couple of them but new errors kept appearing.

Is this something someone else has experienced and know if there are any plans to update the [linux-poplar](https://github.com/96boards-poplar/linux/tree/poplar-4.9) fork. I suspect not as it has been pretty dead for 2 years.

I am trying to re-run the build with:

```
GCCVERSION = "7.%"
```

**Comment by fboudra** [comment]
@petegriffin any infos?

**Comment by mirzak** [comment]
By the way, downgrading the gcc version to 7 "worked".

**Comment by petegriffin** [comment]
@mirzak  @fboudra  I think as a temporary workaround we should force this recipe to build with gcc 7. I have also pinged a mail to the person in HiSilicon and Linaro who was working on this to see if there are any plans to update things.

**Comment by shawnguo2** [comment]
@mirzak Unfortunately, there is no plan to update linux-poplar, so you will have to live with GCC7.x or migrate to GCC8.x on your own.

**Comment by mirzak** [comment]
Thanks for the quick feedback.

That raises the question if the `poplar` support should be dropped in meta-96boards/thud and newer?

Or as suggested force gcc7 build of `linux-poplar`, I would create a PR but it is not clear to me if it is possible to specify gcc version per package in Yocto. 

During my tests I set it globally.

**Comment by petegriffin** [comment]
For Chromium (which officially only supports clang, and frequently breaks with gcc), I use the following patch to force just that recipe to compiled with clang https://github.com/petegriffin/meta-browser/commit/8748a31861404f5b5850201d2aea934bf71a2689. Hopefully something similar can be done for gcc7

**Comment by mirzak** [comment]
Thanks, I will give it a go and report back.

**Comment by daniel-thompson** [comment]
On Wed, May 15, 2019 at 12:48:24AM -0700, Mirza Krak wrote:
> Thanks for the quick feedback.
> 
> That raises the question if the `poplar` support should be dropped in meta-96boards/thud and newer?

AFAICT there's already a PR to fix this in the 96boards-poplar/linux
project:
https://github.com/96boards-poplar/linux/pull/3/commits

It's fine to no longer develop the kernel but it is a shame we cannot
find a way to park the poplar kernel in a "patches welcome" state.

**Comment by shawnguo2** [comment]
> On Wed, May 15, 2019 at 12:48:24AM -0700, Mirza Krak wrote: Thanks for the quick feedback. That raises the question if the `poplar` support should be dropped in meta-96boards/thud and newer?
> AFAICT there's already a PR to fix this in the 96boards-poplar/linux project: https://github.com/96boards-poplar/linux/pull/3/commits It's fine to no longer develop the kernel but it is a shame we cannot find a way to park the poplar kernel in a "patches welcome" state.

Agreed.  Not sure how I missed the PR.  I will handle it tomorrow.  Thanks for pointing me to the PR, @daniel-thompson

**Comment by shawnguo2** [comment]
@mirzak I tested and merged https://github.com/96boards-poplar/linux/pull/3

**Comment by mirzak** [comment]
Thanks @shawnguo2, I have also tested it and it worked well for me as well.

I will create a PR to bump the `linux-poplar` recipe.

**Comment by mirzak** [comment]
https://github.com/96boards/meta-96boards/pull/312 created. Though notice that there seems not have been a revision bump in a while so there are more changes then discussed here.

I have boot tested the change
